### PR TITLE
Simplify build lib

### PR DESCRIPTION
**Pull request by lvh**
Refs #343.

### CODE REVIEW

File: script/lib.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ get_branch() {
   # See http://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables
   # for details about default Travis Environment Variables and their values
   if [ -z "${TRAVIS_BRANCH-}" ]; then
-    BRANCH=$(git symbolic-ref HEAD | sed -e 's,.*/\(.*\),\1,')
+    BRANCH=$(git rev-parse --abbrev-ref HEAD)
```
Comment (lvh) [inline script/lib.sh]: This has two advantages:

1. shorter and look ma, no `sed` ;)
2. better behavior when not on a branch (`HEAD` vs some random error that I think sed will eat)

Works in git 1.7+, which is like from 2010.

File: script/lib.sh [hunk 2]
```diff
@@ -28,23 +28,10 @@ get_commit() {
...
-      if [ "${OLD_TAG}" == "" ]; then
-          VERSION="dev"
-      fi
-  else
-      # We have ourselves a *real* release
-      VERSION=${TRAVIS_TAG}
-  fi
+  # Version will be the most recent tag + "-dev" if working tree is dirty
+  # e.g.: "1.0.1"" or "1.0.1-dev"
+  VERSION=$(git describe --tags --dirty='-dev' 2> /dev/null)
```
Comment (lvh) [inline script/lib.sh]: I'm not actually sure if this ever errors, but I guess it won't hurt.

Comment (rgbkrk) [inline script/lib.sh]: Why not keep the Travis tag when it's defined still?

Comment (lvh) [inline script/lib.sh]: This should work either way (it works as long as the correct commit is checked out), and it's shorter.

**Comment by rgbkrk** [comment]
Time to see how it goes.

**Pull request merged**